concepts/05 UI Components/DataGrid/20 Editing/45 Disable Editing/30 Row.md

```diff
@@ -0,0 +1,210 @@
+If you want to disable data editing within a specific row, take the following steps:
+
+1. Handle the DataGrid's [onEditingStart](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditingStart) event. 
+2. Check the **data** parameter's values. If you want don't want users to edit this row, assign **true** to **e.cancel**.
+
+---
+##### jQuery
+
+    <!-- tab: index.js -->
+    $(function() {
+        $("#dataGrid").dxDataGrid({
+            // ...
+            onEditingStart(e) {
+                e.cancel = e.data.Position === "CEO";
+            }
+        });
+    });
+
+##### Angular
+
+    <!-- tab: app.component.html -->
+    <dx-data-grid 
+        // ...
+        (onEditingStart)="onEditingStart($event)"
+    >   
+    </dx-data-grid>
+
+    <!-- tab: app.component.ts -->
+    import { DxDataGridModule } from "devextreme-angular";
+    // ...
+    export class AppComponent {
+        onEditingStart(e) {
+            e.cancel = e.data.Position === "CEO";
+        }
+    }
+    @NgModule({
+        imports: [
+            // ...
+            DxDataGridModule
+        ],
+        // ...
+    })
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxDataGrid 
+            @editing-start="onEditingStart"
+        >
+        </DxDataGrid>
+    </template>
+
+    <script>
+    import {
+        DxDataGrid,
+        // ...
+    } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            // ...
+        },
+        methods: {
+            onEditingStart(e) {
+                e.cancel = e.data.Position === "CEO";
+            }
+        }
+        // ...
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React, { useCallback } from "react";
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DataGrid } from 'devextreme-react/data-grid';
+
+    function App() {
+        // ...
+        const onEditingStart = useCallback((e) => {
+           e.cancel = e.data.Position === "CEO";
+        }, []);
+
+        return (
+            <DataGrid 
+                onEditingStart={onEditingStart}
+            >
+            </DataGrid>
+        );
+    }
+
+    export default App;
+
+---
+
+The DataGrid's [editing mode](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#mode) defines whether to display the **Edit** button opposite every row. To hide this button for the required rows, use the button's [visible](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/buttons/#visible) property. Assign a function to this property and check a row's data in a similar manner as in the DataGrid's [onEditingStart](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditingStart) event.
```

Comment:
what do you mean by required row?
when you say assign a function, what function to do you mean?
"similar manner" - can you specify?

Reply:
**what do you mean by required row?** - It is the rows, for which you want to hide the Edit buttons

**when you say assign a function, what function to do you mean?** - this function with condition is described in the code snippets below: 

visible: function(e) {
      return e.row.data.Position !== "CEO";
}

**"similar manner" - can you specify?** - there is the link to code snippets, where users can see the "similar manner", if the code snippets below are not clear for them

Reply:
```suggestion
The DataGrid's [editing mode](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#mode) defines whether to display the **Edit** button opposite every row. To hide this button, use the button's [visible](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/buttons/#visible) property. Assign the [PLEASE PROVIDE NAME OF FUNCTION] function to this property and ...
```
"check row data in a similar manner as in the DataGrid's [onEditingStart](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditingStart) event.
When you say "check", do you mean review? Are you trying to say review DataGrid's [onEditingStart](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditingStart) event for similar functionality?

Reply:
Ok! There is the **visible** property.  You can set it to _boolean_ value or assign the callback function to this property. In our case, the callback function is anonymous, so I cannot provide its name.

It is the example:

visible: function(e) { _// it is anonymous function, that returns a boolean value!_
     return e.row.data.Position !== "CEO"; // it is the row data the user need to check!
}

Text can be as follow: The DataGrid's editing mode defines whether to display the Edit button opposite every row. To hide this button, use the button's visible property. Assign a callback to this property and check row's data within this function.

Reply:
DataGrid's editing mode defines whether to display an Edit button opposite every row. To hide this button, use the button's visible property. Assign a callback to this property and check row data within this function.